tech-debt: SQLite file-handle leak causes TempDir cleanup races on Windows

## Symptom

Two tests in `test/unit/controller/` fail intermittently on Windows local + race-enabled runs with:

\`\`\`
TempDir RemoveAll cleanup: unlinkat C:\Users\cfg\AppData\Local\Temp\1\TestXxx\001\cfgms.db: The process cannot access the file because it is being used by another process.
\`\`\`

Repros consistently:
- TestModuleInterface
- TestModuleCustomBehavior

\`\`\`
$ go test -race -short -timeout 60s ./test/unit/controller/ -run "TestModuleInterface|TestModuleCustomBehavior"
--- FAIL: TestModuleInterface (3.36s)
--- FAIL: TestModuleCustomBehavior (2.81s)
\`\`\`

## Root cause

The test creates a SQLite DB inside a t.TempDir() and exercises a controller stack that opens the DB. When the test goroutine returns, t.TempDir's deferred cleanup runs and tries to delete the .db file before the SQLite connection has been closed and Windows has released the file handle.

POSIX rename/remove semantics allow this (orphaned-inode); NTFS does not — the file remains locked until the holder closes.

## Why it doesn't show on Linux CI

POSIX file semantics; cleanup succeeds even with open handles.

## Fix shape

Either:
1. Add explicit \`db.Close()\` (or a stack-shutdown helper) to the test before TempDir cleanup runs.
2. Use a t.Cleanup() that closes the stack BEFORE the implicit TempDir cleanup.
3. Add a small retry loop around \`os.RemoveAll\` in test helpers for Windows.

Option 2 is cleanest. The test fixture in test/unit/controller/setup_test.go likely needs a Close() helper that test functions call (or register via t.Cleanup) before returning.

## Discovery

Surfaced during Story #1918 (steward launcher) push-readiness validation. The launcher itself doesn't touch this code; failure is in unrelated files (\`test/unit/controller/\` last touched in PR #1785, #722, #432).

--- a/test/unit/controller/module_test.go
+++ b/test/unit/controller/module_test.go
@@ -39,6 +39,18 @@ func newTestController(t *testing.T) *controller.Controller {
 	logger := unit.NewTestLogger(t)
 	ctrl, err := controller.New(cfg, logger)
 	require.NoError(t, err)
+
+	// Close the controller stack before the deferred t.TempDir() cleanup runs.
+	// t.Cleanup callbacks run LIFO and before TempDir removal, so closing here
+	// releases the SQLite file handle and avoids the Windows "file is being used
+	// by another process" unlink error during TempDir cleanup (#1923). Cleanup
+	// errors are secondary to the test outcome, so log rather than fail.
+	t.Cleanup(func() {
+		if cerr := ctrl.Close(); cerr != nil {
+			t.Logf("newTestController: controller close during cleanup: %v", cerr)
+		}
+	})
+
 	return ctrl
 }
 

